wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -661,30 +751,32 @@ instance ToSchema TransactionDirection where
     declareNamedSchema _ =
         pure $ NamedSchema (Just "TransactionDirection") $ mempty
             & type_ .~ SwaggerString
-            & enum_ .~ Just ["outgoing", "incoming"]
+            & enum_ ?~ ["outgoing", "incoming"]
 
 -- | A 'Wallet''s 'Transaction'.
 data Transaction = Transaction
   { txId            :: !(V1 Core.TxId)
-    -- ^ The Tx Id.
   , txConfirmations :: !Word
-    -- ^ The number of confirmations.
   , txAmount        :: !(V1 Core.Coin)
-    -- ^ The 'Coin' moved as part of this transaction.
   , txInputs        :: !(NonEmpty PaymentDistribution)
-    -- ^ The input money distribution.
   , txOutputs       :: !(NonEmpty PaymentDistribution)
-    -- ^ The output money distribution.
   , txType          :: TransactionType
-    -- ^ The type for this transaction (e.g local, foreign, etc).
   , txDirection     :: TransactionDirection
-    -- ^ The direction for this transaction (e.g incoming, outgoing).
   } deriving (Show, Ord, Eq, Generic)
 
 deriveToJSON Serokell.defaultOptions ''Transaction
 
 instance ToSchema Transaction where
-    declareNamedSchema = genericSchemaDroppingPrefix "tx"
+  declareNamedSchema =
+    genericSchemaDroppingPrefix "tx" (\(--^) props -> props
+      & ("id"            --^ "Transaction's id")
```

Comment:
This would be probably overkill, but I have seen some ideas floating around in the Servant issue tracker where people were thinking of using a type similar to `(--^)` to attach the documentation stuff directly in the types, when you declare the data record. Example:

``` haskell
data Foo = Foo {
  bar :: Int --^ "Documentation for Int"
}
```

It probably makes working with these types madness, in practice, so I'm happy you went for a different approach, but I thought I would share it just for reference. Incidentally, I think also Gabriel Gonzalez uses a similar combinator is his library `optparse-generic`.

Reply:
I've tried this approach on a previous project. Very neat to do generic stuff with the types after that, yet a bit painful indeed because each field has to be unwrap after that; compile-times are also significantly slower. 

Note that unfortunately, I didn't find any way to have this statically checked without altering or annotating the original data-type. It will however throw explicitly should we try to document a non-existing field.

Reply:
TIL you can introduce operators in a lambda -- super cool idiom :smile:

Reply:
Haha, I believe that's the first time I do it... but it didn't felt wrong, and it compiled so... ^.^